src/component.jl

```diff
@@ -2,7 +2,12 @@ function add_forecast!(
                        component::T,
                        forecast::ForecastInternal,
                       ) where T <: InfrastructureSystemsType
-    add_forecast!(_get_forecast_container(component), forecast)
+    container = _get_forecast_container(component)
+    if isnothing(container)
+        throw(ArgumentError("type $T does not contain forecasts"))
```

Comment:
The message isn't clear if the cause of the error is the type $T not having the forecast field or the data not being available. 

I would suggest 
```Julia
throw(ArgumentError("type $T can not store forecasts"))
```

Reply:
Fixed